### ISSUE TITLE
Incorrect working of $third-party modifier in Safari

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdGuardForSafari#104]
@Alex-302 commented on [Wed Jul 20 2016](https://github.com/AdguardTeam/AdguardBrowserExtension/issues/297)

Sample page:
https://www.tripadvisor.com/Hotel_Review-g60811-d93892-Reviews-Home2_Suites_by_Hilton_Baltimore_Downtown-Baltimore_Maryland.html
Images are blocked by the rule from Easylist China

```
||tripadvisor.com^$third-party
```

Image sample:
https://media-cdn.tripadvisor.com/media/photo-s/0b/b3/b1/a3/photo4jpg.jpg
This rule should not works on tripadvisor.com domain.
Screenshots:
<details>

![image](https://cloud.githubusercontent.com/assets/15856304/16941178/8b07c3cc-4d5c-11e6-8410-828286a1fae9.png)

![image](https://cloud.githubusercontent.com/assets/15856304/16941188/95e21176-4d5c-11e6-9f70-3b590d5fb34f.png)
</details>

---
fixed one screenshot

**Comment by tomahl** [comment]
This is still not working as expected. For now it can be solved by using `||example.org^$third-party,domain=~example.org`

Why is this ticket being postponed over and over?

**Comment by vozersky** [comment]
We shall re-test it before pushing 1.9 beta

**Comment by ameshkov** [comment]
> Why is this ticket being postponed over and over?

The fix is trivial indeed, but the problem is that there are too many `$third-party` rules in the filter lists. Rewriting them all to `||example.org^$domain=~example.org` will increase the compilation time (Safari already takes too long to consume content blockers) and increase memory usage. We should be careful with that.
